addons/website/models/ir_http.py

```diff
@@ -41,6 +41,23 @@ def sitemap_qs2dom(qs, route, field='name'):
     return dom
 
 
+def get_request_website():
+    """ Return the website set on `request` if called in a frontend context
+    (website=True on route).
+    This method can typically be used to check if we are in the frontend.
+
+    This method is easy to mock during python tests to simulate frontend
+    context, rather than mocking every method accessing request.website.
+
+    Don't import directly the method or it won't be mocked during tests, do:
+    ```
+    from odoo.addons.website.models import ir_http
+    my_var = ir_http.get_request_website()
+    ```
+    """
+    return request and getattr(request, 'website', False) or False
```

Comment:
I agree but since it is not a model method I can't easily return an empty recordset.
And we don't want it to be a website model method as we want to keep it on http.
So as discussed, for now we will keep False as possible return value.